x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/threshold/find_threshold_signals.ts

```diff
@@ -142,30 +145,32 @@ export const findThresholdSignals = async ({
       secondaryTimestamp,
     });
 
-    const searchResultWithAggs = searchResult as ThresholdSingleBucketAggregationResult;
-    if (!searchResultWithAggs.aggregations) {
-      throw new Error('Aggregations were missing on threshold rule search result');
-    }
-
     searchAfterResults.searchDurations.push(searchDuration);
     searchAfterResults.searchErrors.push(...searchErrors);
 
-    const docCount = searchResultWithAggs.hits.total.value;
```

Comment:
nit: would prefer early return here, upon checking this condition

```
      !searchResultHasAggs<ThresholdSingleBucketAggregationResult>(searchResult) &&
      isEmpty(searchErrors)
```

that would help to avoid nested `if` conditions:
>     if (searchResultHasAggs<ThresholdSingleBucketAggregationResult>(searchResult)) {
>>   if (
        docCount >= threshold.value &&
        (!includeCardinalityFilter ||
          (searchResult?.aggregations?.cardinality_count?.value ?? 0) >=
            threshold.cardinality[0].value)